pages/apis/webhooks.md.erb

```diff
@@ -63,6 +64,39 @@ Each event’s data is sent JSON encoded in the request body. See each event’s
   <p>Note that if a builds transitions between states very quickly, for example from blocked (`finished`) to unblocked (`running`), the webhook may be in a different state from the actual build. This is a known limitation of webhooks, in that they may represent a later version of the object than the one that triggered the event.</p>
 </div>
 
+## Verifying Webhook signatures
+
+The `X-Buildkite-Signature` header included in each signed event contains a timestamp and one signature. The timestamp is prefixed by `timestamp=`, and each signature is prefixed by `signature=`.
+
+Buildkite generates signature using a hash-based message authentication code ([HMAC](https://en.wikipedia.org/wiki/HMAC)) with [SHA-256](https://en.wikipedia.org/wiki/SHA-2). The timestamp is an integer represents UTC timestamp.
+
+Example using Ruby:
+
+```ruby
+class BuildkiteWebhook
+  def self.valid?(payload, header, secret)
+    timestamp, signature = get_timestamp_and_signatures(header)
+    expected = OpenSSL::HMAC.hexdigest("sha256", secret, "#{timestamp}.#{payload}")
```

Comment:
I know this is only an example in a single language so it's not intended to be a copy-pastable thing folks can use, but maybe we want to include a `require 'openssl'` at the top?